### PR TITLE
Initial Update README.md + CONTRIBUTING.md

### LINKED ISSUE #692: Update / Improve README.md
Our [README.md](https://github.com/nwnxee/unified/blob/master/README.md) has a bunch of outdated info, we should update it!

Also up for consideration is moving the [Contributing code](https://github.com/nwnxee/unified/blob/master/README.md#contributing-code) stuff to its own CONTRIBUTING.md and adding the NWScript style info.

### PR DESCRIPTION
**Pull request by Daztek**
Resolves #692 

Initial update, suggestions and comments welcome!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,19 +20,19 @@ NWNX:EE is not just for plugin developers. Module creators can use plugins creat
 
 ## What versions of the server are supported?
 
-The most recent build of NWN:EE that is supported can be discovered at the top of this file. Please note that we do NOT support any server binaries other than the one in the dedicated server package. We support neither the binary released through the Beamdog client nor the one released through the Steam client. We release an update for each new version of the dedicated server package, which can be found here: https://forums.beamdog.com/discussion/67157/server-download-packages-and-docker-support/p1
+The most recent build of NWN:EE that is supported can be discovered at the top of this file. Please note that we do NOT support any server binaries other than the one in the dedicated server package. We support neither the binary released through the Beamdog client nor the one released through the Steam client. We release at least one update for each new version of the dedicated server package, which can be found here: [Server download packages and docker support](https://forums.beamdog.com/discussion/67157/server-download-packages-and-docker-support/p1)
```
Comment (mtijanic) [inline README.md]: Add a note that the steam and BDC binaries are almost always identical to the one we support. We can add an md5sum of the binary we support to the readme.

File: README.md [hunk 2]
```diff
@@ -17,19 +20,19 @@ NWNX:EE is not just for plugin developers. Module creators can use plugins creat
...
 
 ## What platforms are supported?
 
-Only Linux is currently supported. It is possible to run NWNX:EE on Windows using Docker to emulate a Linux server environment. Refer to the sections below to find out how. Many discussions have taken place regarding work on a native Windows port of NWNX:EE. It has been tackled several times in the past, but none of those attempts have seen fruition - complications have arisen in the course of the work. We know how we would do it, and estimate the time required for the task to be between 2 and 4 weeks of full-time work, but nobody with the skillset necessary to take on the project has the time or interest at the moment to do so. Therefore, users should not expect a native Windows port from a member of the team in the near future. (contributions welcome!)
+Only Linux is natively supported. It is possible to run NWNX:EE on Windows using Docker to emulate a Linux server environment or WSL/WSL2 when using Windows 10. At this point in time there will be no native Windows support for  NWNX:EE.
```
Comment (mtijanic) [inline README.md]: `s/there will be no/there are no plans for/`

File: README.md [hunk 3]
```diff
@@ -17,19 +20,19 @@ NWNX:EE is not just for plugin developers. Module creators can use plugins creat
...
 
 ## How do I use it?
 
 First of all, you need to get the NWNX binaries. If you want to compile the binaries yourself, refer to the compiling section below. Otherwise, you can download the binaries by clicking on the [Releases](https://github.com/nwnxee/unified/releases) tab at the top of the GitHub repository page. Keep in mind these are generally only updated whenever a new NWN patch is released.
 
 ## Integrating NWNX:EE into your module
 
-The built binaries (when built using the build script or downloaded as artifacts) will contain a folder in which all of the script files associated with NWNX:EE live. Simply import these into your module and call the functions as you would any other scripting function. Some may not compile with the default compiler - it is safe to delete these scripts if you do not need to use the associated plugins.
+Each NWNX:EE [Release](https://github.com/nwnxee/unified/releases) has a `NWScript.zip` file containing all of the scripts files, if you have built NWNX:EE yourself the `NWScript.zip` file will be in your `Binaries` folder. Simply import the scripts into your module and call the functions as you would any other scripting function.
```
Comment (mtijanic) [inline README.md]: "Be sure to update these scripts whenever updating NWNX, and then rebuild all your scripts in your module."

File: README.md [hunk 4]
```diff
@@ -57,7 +60,7 @@ Optionally, you can use Docker to run the server with pre-built NWNX binaries. R
 
 ## Running the server (Docker)
 
-`nwnxee/unified` supports the following [tags](https://cloud.docker.com/repository/docker/nwnxee/unified/tags):
+`nwnxee/unified` supports the following [tags](https://hub.docker.com/r/nwnxee/unified/tags):
```
Comment (mtijanic) [inline README.md]: We also support `:buildXXXX`, no?

File: README.md [hunk 5]
```diff
@@ -78,9 +81,13 @@ It is advised to configure the container with docker-compose. Please refer to th
...
 * Example compose file: [nwnxee-template](https://github.com/Urothis/nwnxee-template/blob/master/docker-compose.yml)
 
+## Running the server (Windows 10 WSL1)
+
+As of NWN:EE Release 8193 it is possible to run `nwserver-linux` and NWNX:EE on Windows 10's Subsystem for Linux. See this guide on how to do so: [NWNX:EE on WSL1, a brief guide](https://github.com/Daztek/NWNX-WSL)
+
 ## Compiling NWNX:EE (native)
 
-Please note that you need a modern C++ compiler to build NWNX:EE. The code is confirmed to build on GCC 6.4 + or Clang 4.0 +. It may be possible to build on an older compiler.
+Please note that you need a modern C++ compiler to build NWNX:EE. The code is confirmed to build on GCC 7.4+ or Clang 5.0+. It may be possible to build on an older compiler.
```
Comment (mtijanic) [inline README.md]: "modern C++17 compatible compiler"

File: README.md [hunk 6]
```diff
@@ -90,7 +97,7 @@ To build the typical way:
 
 Or, if you want to compile manually:
 
-1.  Execute CC=gcc -m32 CXX=g++ -m32 mkdir build && cd build && cmake .. && make
+1.  Execute CC=gcc -m64 CXX=g++ -m64 mkdir build-nwnx && cd build-nwnx && cmake .. && make
```
Comment (mtijanic) [inline README.md]: `-m64` is not necessary when building on 64bit machines, and thus setting `CC` and `CXX` is also not. So just `mkdir && cd && cmake && make`

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+## How to contribute to NWNX:EE
```
Comment (mtijanic) [inline CONTRIBUTING.md]: This section needs some work, but I'll do that. Good enough for now.

**Comment by Daztek** [comment]
Updated

**Pull request merged**